controller/string_cache.go

```diff
@@ -0,0 +1,28 @@
+package controller
+
+import (
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+)
+
+type stringCache struct {
+	underlying *gocache.Cache
+}
+
+func newStringCache(expiration time.Duration) *stringCache {
+	c := &stringCache{
+		underlying: gocache.New(expiration, expiration/2),
```

Comment:
The cleanup interval is now 150% of the resync period, because (resync period * 3) / 2. Why is that?

Reply:
I didn't think about resync period when writing that. This is my thinking was:

- It should be fine to run cleanup twice during the expiration of an object. This cache shouldn't be too heavily used so it should be pretty empty w/o running cleanup and blocking the cache too often.
- Re tripple resync period expiration - I just thought it should be more than enough to keep information for so long. In general it should be enough to keep it for single reconciliation loop. If the loop blocks for longer than single resync period there will be other problems like controller choking.